lib/instrumentation/index.js

```diff
@@ -76,11 +110,29 @@ Instrumentation.prototype.start = function () {
     require('./patch-async')(this)
   }
 
+  const patches = this._agent._conf.addPatch
+  if (Array.isArray(patches)) {
+    for (let [mod, path] of patches) {
+      this.addPatch(mod, path)
+    }
+  }
+
+  this._startHook()
+}
+
+Instrumentation.prototype._startHook = function () {
+  if (!this._started) return
+  if (this._hook) {
+    this._agent.logger.debug('removing hook to Node.js module loader')
+    this._hook.unhook()
```

Comment:
I'm not sure that we can support unhooking and the adding hooks again 🤔 I think that all calls to `require()` that have already been called, would not get updated with the new hooks.

Reply:
It wouldn't impact past requires, but we already have that with the timing of the `agent.start()` call, so I don't think that really matters here. I think it's fine that a patch only applies to future require calls. The target use case for the patch registry interface would be to use it immediately after `agent.start()`, before anything else is loaded.